skbio/io/registry.py

```diff
@@ -422,8 +422,14 @@ def sniff(self, file, **kwargs):
 
         return matches[0]
 
-    def _find_matches(self, file, lookup, **kwargs):
+    def _find_matches(self, file, lookup, into=None, **kwargs):
         matches = []
+        # Only run sniffers for the relevant formats for the given skbio object.
+        if into is not None:
```

Comment:
I am thinking if this code block can be elevated into `sniff`, such that we don't have to override a mandatory parameter (`lookup`)?